command/kv_patch.go

```diff
@@ -211,6 +223,15 @@ func (c *KVPatchCommand) Run(args []string) int {
 		return 2
 	}
 
+	// collecting data to be removed
+	if newData == nil {
+		newData = make(map[string]interface{})
+	}
+
+	for _, key := range c.flagRemoveData {
+		newData[key] = nil
```

Comment:
Not sure if we wanted to include the same comment that you had above in the `kv metadata patch` implementation mentioning why the `nil` is important. What do you think?